### PR TITLE
make ChainRulesCore dependency into an extension

### PR DESCRIPTION
**Pull request by KristofferC**
Shouldn't be merged until the Registrator deployment of https://github.com/JuliaRegistries/Registrator.jl/pull/397 is done.

### CODE REVIEW

**Comment by stevengj** [comment]
Could you summarize the practical effect for those of us who have not followed the "weak dependency" feature progress?

**Comment by KristofferC** [comment]
Sure, I'll summarize but also link to the docs: https://pkgdocs.julialang.org/dev/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions)

Basically, you can think of `ext/ChainRulesCore.jl` as a "package" that gets automatically loaded when `ChainRulesCore` gets loaded. So "extensions" allow adding e.g. method extensions to packages without having to unconditionally load them (which can be expensive from a load time p.o.v).

File: src/SpecialFunctions.jl [hunk 1]
```diff
@@ -91,7 +90,9 @@ include("gamma.jl")
 include("gamma_inc.jl")
 include("betanc.jl")
 include("beta_inc.jl")
-include("chainrules.jl")
+if !isdefined(Base, :get_extension)
```
Comment (giordano) [inline src/SpecialFunctions.jl:93]: Do I understand correctly this PR is basically removing functionality for Julia v1.8-?

Comment (KristofferC) [reply src/SpecialFunctions.jl:93]: That'd be pretty unfortunate :p.

**Comment by devmotion** [comment]
Seems the PR is ready and the Registrator deployment is done?

**Comment by KristofferC** [comment]
Yes!

**Review by devmotion — APPROVED** [review]

File: src/SpecialFunctions.jl (hunk 1 above)
Comment (stevengj) [reply src/SpecialFunctions.jl:93]: for Julia 1.8- my understanding is that it is the same as the old behavior (i.e. it loads ChainRulesCore unconditionally?)
